extensions/notebook/src/book/bookModel.ts

```diff
@@ -334,6 +334,10 @@ export class BookModel {
 		return treeItems;
 	}
 
+	private setHierarchyId(iteration: number, hierarchyId?: string): string {
```

Comment:
This would be clearer as `generateHierarchyId`

I would also suggest something like `sectionNumber` instead of `iteration` since that's pretty generic. (or even better find some name indicating what it's used for in the generated hierarchy id...)

Reply:
maybe like a sectionIndex instead